be/src/vec/functions/function_case.h

```diff
@@ -323,7 +324,9 @@ class FunctionCase : public IFunction {
         size_t rows_count = column_holder.rows_count;
         result_column_ptr->resize(rows_count);
         auto* __restrict result_raw_data =
-                assert_cast<ColumnType*>(result_column_ptr.get())->get_data().data();
```

Comment:
not need?

Reply:
result_column_ptr is constructed at https://github.com/apache/doris/pull/39030/files/2c7c1a234dceed3f79aca4e5081bf47a6b18bfc6#diff-f583c787d6c6f1e0d97ef69da94a68e5eaca04d4c07aae9d9920a2eae7d1da49L255,  type safe seems has already been guaranteed

Reply:
> result_column_ptr is constructed by `auto result_column_ptr = data_type->create_column();`, type safe seems has already been guaranteed

+1